src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -487,6 +482,31 @@ private String resolveContent(String result, Set<String> usedIds) {
     }
 
     /**
+     * Resolves the refLabel with the given information using reSolveString(...). If the ref label could be resolved,
+     * then the resolved string will be add to newRes. Otherwise the refLabel string will be added to newRes.
+     *
+     * @param res String to resolve
+     * @param newRes StringBuilder to append the labels
+     * @param next ref label position as int
+     * @param stringEnd end position of the label as int
+     * @param usedIds Set of Strings, containing the IDs that are used
+     */
+    private void resolveRefLabel(String res, StringBuilder newRes, int next, int stringEnd, Set<String> usedIds) {
+        String refLabel = res.substring(next + 1, stringEnd);
+        String resolved = resolveString(refLabel, usedIds);
+
+        if (resolved == null) {
+            // Could not resolve string. Display the #
+            // characters rather than removing them:
+            newRes.append(res.substring(next, stringEnd + 1));
+        } else {
+            // The string was resolved, so we display its meaning only,
+            // stripping the # characters signifying the string label:
+            newRes.append(resolved);
```

Comment:
I think those comments should be moved to the javadoc comment of this method or they should be removed.